server/app/services/question/types/QuestionDefinitionConfig.java

```diff
@@ -0,0 +1,75 @@
+package services.question.types;
+
+import com.google.auto.value.AutoValue;
+import java.time.Instant;
+import java.util.Optional;
+import java.util.OptionalLong;
+import services.LocalizedStrings;
+
+@AutoValue
+public abstract class QuestionDefinitionConfig {
+
+  abstract String name();
+
+  abstract String description();
+
+  // Note: you must check prefixes anytime you are doing a locale lookup
+  // see getQuestionText body comment for explanation.
+  abstract LocalizedStrings questionText();
+
+  abstract LocalizedStrings questionHelpText();
+
+  abstract QuestionDefinition.ValidationPredicates validationPredicates();
+
+  abstract OptionalLong id();
+
+  abstract Optional<Long> enumeratorId();
+
+  abstract Optional<Instant> lastModifiedTime();
+
+  public static RequiredName builder() {
```

Comment:
suggestion (non-blocking): Is the step builder pattern used elsewhere in the code base? If not, it could be helpful to link to the documentation here, or provide a longer comment explaining how they work.

Reply:
Yes, it's also used in `Modal.java`:

https://github.com/civiform/civiform/blob/69750e64b0bdc1d4b6a9b97f70f820fa72a40fa3/server/app/views/components/Modal.java#L19-L81